metadata-builders/src/test/java/com/linkedin/metadata/builders/search/DatasetIndexBuilderTest.java

```diff
@@ -1,40 +1,265 @@
+  private final DatasetIndexBuilder _indexBuilder = new DatasetIndexBuilder();
 
   @Test
-  public void testDescriptionClearing() {
-    DatasetUrn datasetUrn = new DatasetUrn(new DataPlatformUrn("foo"), "bar", FabricType.PROD);
-
-    DatasetProperties datasetProperties = new DatasetProperties().setDescription("baz");
-    DatasetSnapshot datasetSnapshot = ModelUtils.newSnapshot(DatasetSnapshot.class, datasetUrn,
+  public void datasetProperties_setsDescription() {
+    // given
+    final DatasetUrn datasetUrn = new DatasetUrn(new DataPlatformUrn("foo"), "bar", FabricType.PROD);
+    final DatasetProperties datasetProperties = new DatasetProperties().setDescription("baz");
+    final DatasetSnapshot datasetSnapshot = ModelUtils.newSnapshot(DatasetSnapshot.class, datasetUrn,
         Collections.singletonList(ModelUtils.newAspectUnion(DatasetAspect.class, datasetProperties)));
-    List<DatasetDocument> actualDocs = new DatasetIndexBuilder().getDocumentsToUpdate(datasetSnapshot);
-    assertEquals(actualDocs.size(), 1);
-    assertEquals(actualDocs.get(0).getUrn(), datasetUrn);
-    assertEquals(actualDocs.get(0).getDescription(), "baz");
+    final DatasetDocument expectedDocument = new DatasetDocument().setUrn(datasetUrn)
+        .setBrowsePaths(new StringArray("/prod/foo/bar"))
+        .setOrigin(FabricType.PROD)
+        .setName("bar")
+        .setPlatform("foo")
+        .setDescription("baz");
+
+    // when
+    final List<DatasetDocument> actualDocs = _indexBuilder.getDocumentsToUpdate(datasetSnapshot);
+
+    // then
+    assertThat(actualDocs).containsExactly(expectedDocument);
+  }
 
-    datasetProperties = new DatasetProperties();
-    datasetSnapshot = ModelUtils.newSnapshot(DatasetSnapshot.class, datasetUrn,
+  @Test
+  public void datasetProperties_noDescription() {
+    // given
+    final DatasetUrn datasetUrn = new DatasetUrn(new DataPlatformUrn("foo"), "bar", FabricType.PROD);
+    final DatasetProperties datasetProperties = new DatasetProperties();
+    final DatasetSnapshot datasetSnapshot = ModelUtils.newSnapshot(DatasetSnapshot.class, datasetUrn,
         Collections.singletonList(ModelUtils.newAspectUnion(DatasetAspect.class, datasetProperties)));
-    actualDocs = new DatasetIndexBuilder().getDocumentsToUpdate(datasetSnapshot);
-    assertEquals(actualDocs.size(), 1);
-    assertEquals(actualDocs.get(0).getUrn(), datasetUrn);
-    assertEquals(actualDocs.get(0).getDescription(), "");
+    final DatasetDocument expectedDocument = new DatasetDocument().setUrn(datasetUrn)
+        .setBrowsePaths(new StringArray("/prod/foo/bar"))
+        .setOrigin(FabricType.PROD)
+        .setName("bar")
+        .setPlatform("foo")
+        .setDescription("");
+
+    // when
+    final List<DatasetDocument> actualDocs = new DatasetIndexBuilder().getDocumentsToUpdate(datasetSnapshot);
+
+    // then
+    assertThat(actualDocs).containsExactly(expectedDocument);
+  }
+
+  @Test
+  public void datsetDeprecation() {
```

Comment:
nit: `datasetDeprecation` not `datsetDeprecation`?
Other places in this file have similar typos.